Drop-In/src/main/java/com/braintreepayments/api/PaymentMethodInspector.java

```diff
@@ -0,0 +1,112 @@
+        if (paymentMethodNonce instanceof CardNonce) {
+            return ((CardNonce) paymentMethodNonce).getLastFour();
+        } else if (paymentMethodNonce instanceof PayPalAccountNonce) {
+            return ((PayPalAccountNonce) paymentMethodNonce).getEmail();
+        } else if (paymentMethodNonce instanceof VenmoAccountNonce) {
+            return ((VenmoAccountNonce) paymentMethodNonce).getUsername();
+        } else if (paymentMethodNonce instanceof GooglePayCardNonce) {
+            return ((GooglePayCardNonce) paymentMethodNonce).getLastFour();
+        } else {
+            return "";
+        }
+    }
+
+    DropInPaymentMethod getPaymentMethod(PaymentMethodNonce paymentMethodNonce) {
+        String canonicalName = getPaymentMethodCanonicalName(paymentMethodNonce);
+        if (canonicalName != null) {
+            switch (canonicalName) {
+                case PAYMENT_METHOD_AMEX:
+                    return DropInPaymentMethod.AMEX;
+                case PAYMENT_METHOD_DINERS_CLUB:
+                    return DropInPaymentMethod.DINERS_CLUB;
+                case PAYMENT_METHOD_DISCOVER:
+                    return DropInPaymentMethod.DISCOVER;
+                case PAYMENT_METHOD_JCB:
+                    return DropInPaymentMethod.JCB;
+                case PAYMENT_METHOD_MAESTRO:
+                    return DropInPaymentMethod.MAESTRO;
+                case PAYMENT_METHOD_MASTERCARD:
+                    return DropInPaymentMethod.MASTERCARD;
+                case PAYMENT_METHOD_VISA:
+                    return DropInPaymentMethod.VISA;
+                case PAYMENT_METHOD_UNION_PAY:
+                    return DropInPaymentMethod.UNIONPAY;
+                case PAYMENT_METHOD_HIPER:
+                    return DropInPaymentMethod.HIPER;
+                case PAYMENT_METHOD_HIPERCARD:
+                    return DropInPaymentMethod.HIPERCARD;
+                case PAYMENT_METHOD_PAYPAL:
+                    return DropInPaymentMethod.PAYPAL;
+                case PAYMENT_METHOD_VENMO:
+                    return DropInPaymentMethod.VENMO;
+                case PAYMENT_METHOD_GOOGLE_PAY:
+                    return DropInPaymentMethod.GOOGLE_PAY;
+                default:
+                    return null;
+            }
+        }
+        return null;
+    }
+
+    private String getPaymentMethodCanonicalName(PaymentMethodNonce paymentMethodNonce) {
+        if (paymentMethodNonce instanceof CardNonce) {
+            return ((CardNonce) paymentMethodNonce).getCardType();
+        } else if (paymentMethodNonce instanceof PayPalAccountNonce) {
+            return PAYMENT_METHOD_PAYPAL;
+        } else if (paymentMethodNonce instanceof VenmoAccountNonce) {
+            return PAYMENT_METHOD_VENMO;
+        } else if (paymentMethodNonce instanceof GooglePayCardNonce) {
+            return PAYMENT_METHOD_GOOGLE_PAY;
+        } else {
+            return null;
+        }
+    }
+
+    CardType parseCardType(String cardType) {
```

Comment:
```suggestion
    CardType getCardType(String cardType) {
```

Another nitpick - but this naming seems a little more straightforward to me, since we aren't really "parsing" anything

Reply:
Yeah def. I'll rename.